src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,429 @@
+    begin
+      induction n with k hk generalizing x y,
+      refl,
+      rw [prod.pow_snd, prod.pow_fst] at hk ⊢,
+      rw [pow_succ a.snd, mul_apply, a.central, hk, pow_succ' a.fst, mul_apply],
+    end⟩, }
+
+@[simp] lemma add_fst (a b : 𝓜(𝕜, A)) : (a + b).fst = a.fst + b.fst := rfl
+@[simp] lemma add_snd (a b : 𝓜(𝕜, A)) : (a + b).snd = a.snd + b.snd := rfl
+@[simp] lemma zero_fst : (0 : 𝓜(𝕜, A)).fst = 0 := rfl
+@[simp] lemma zero_snd : (0 : 𝓜(𝕜, A)).snd = 0 := rfl
+@[simp] lemma neg_fst (a : 𝓜(𝕜, A)) : (-a).fst = -a.fst := rfl
+@[simp] lemma neg_snd (a : 𝓜(𝕜, A)) : (-a).snd = -a.snd := rfl
+@[simp] lemma sub_fst (a b : 𝓜(𝕜, A)) : (a - b).fst = a.fst - b.fst := rfl
+@[simp] lemma sub_snd (a b : 𝓜(𝕜, A)) : (a - b).snd = a.snd - b.snd := rfl
+@[simp] lemma one_fst : (1 : 𝓜(𝕜, A)).fst = 1 := rfl
+@[simp] lemma one_snd : (1 : 𝓜(𝕜, A)).snd = 1 := rfl
+@[simp] lemma mul_fst (a b : 𝓜(𝕜, A)) : (a * b).fst = a.fst * b.fst := rfl
+@[simp] lemma mul_snd (a b : 𝓜(𝕜, A)) : (a * b).snd = b.snd * a.snd := rfl
+@[simp] lemma nat_cast_fst (n : ℕ) : (n : 𝓜(𝕜 , A)).fst = n := rfl
+@[simp] lemma nat_cast_snd (n : ℕ) : (n : 𝓜(𝕜 , A)).snd = n := rfl
+@[simp] lemma int_cast_fst (n : ℤ) : (n : 𝓜(𝕜 , A)).fst = n := rfl
+@[simp] lemma int_cast_snd (n : ℤ) : (n : 𝓜(𝕜 , A)).snd = n := rfl
+@[simp] lemma pow_fst (n : ℕ) (a : 𝓜(𝕜, A)) : (a ^ n).fst = a.fst ^ n := rfl
+@[simp] lemma pow_snd (n : ℕ) (a : 𝓜(𝕜, A)) : (a ^ n).snd = a.snd ^ n := rfl
+
+/-- The natural injection from `double_centralizer.to_prod` except the second coordinate inherits
+`mul_opposite.op`. The ring structure on `𝓜(𝕜, A)` is the pullback under this map. -/
+def to_prod_mop : 𝓜(𝕜, A) → ((A →L[𝕜] A) × (A →L[𝕜] A)ᵐᵒᵖ) :=
+λ a, (a.fst, mul_opposite.op a.snd)
+
+/-- The ring structure is inherited as the pullback under the injective map
+`double_centralizer.to_prod_mop : 𝓜(𝕜, A) → (A →L[𝕜] A) × (A →L[𝕜] A)ᵐᵒᵖ` -/
+instance : ring 𝓜(𝕜, A) :=
+function.injective.ring to_prod_mop
+  (λ a b h, let h' := prod.ext_iff.mp h in ext _ _ $ prod.ext h'.1 $ mul_opposite.op_injective h'.2)
+  rfl rfl (λ _ _, rfl) (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl)
+  (λ x n, by simpa only [to_prod_mop, to_prod, prod.smul_fst, prod.smul_snd, mul_opposite.op_smul])
+  (λ x n, by simpa only [to_prod_mop, to_prod, prod.smul_fst, prod.smul_snd, mul_opposite.op_smul])
+  (λ x n, by simpa only [to_prod_mop, to_prod, prod.pow_fst, prod.pow_fst, mul_opposite.op_pow])
+  (λ _, rfl) (λ _, rfl)
+
+/-- The canonical map `double_centralizer.to_prod` as an additive group homomorphism. -/
+def add_group_hom_to_prod : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A) :=
+{ to_fun := to_prod,
+  map_zero' := rfl,
+  map_add' := λ x y, rfl }
+
+/-- The module structure is inherited as the pullback under the additive group monomoprhism
+`double_centralizer.to_prod : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A)` -/
+instance : module 𝕜 𝓜(𝕜, A) :=
+function.injective.module 𝕜 add_group_hom_to_prod ext (λ x y, rfl)
+
+noncomputable instance : algebra 𝕜 𝓜(𝕜, A) :=
```

Comment:
Not that it's important, but what's the root of noncomputability here? Is it just because you wrote `noncomputable theory` at the top of the file?

Reply:
I can't remember why I put it there, but it doesn't seem necessary now. Removed.

Reply:
Are the other `noncomputable` needed too? Note that you need to remove `noncomputable theory` to decide whether they are or not.